### ISSUE TITLE
build: OSX has no getent

### ISSUE THREAD

**Issue by steveschnepp** [munin-monitoring/munin#451]
Seems OSX has no `getent`

```
$ make clean
which: no getent in ( ... )
```

**Comment by pdvnl** [comment]
OSX has no getent: true but not a problem. The makefile for 2.0.34 has the following lines:
```
# For OSX 10.5 (Leopard), use the following two lines instead of what's above
#
#CHECKUSER  := $(shell dscl . -read /Users/$(USER) >/dev/null 2>/dev/null || (echo "echo User $(USER) nonexistent. Create the user and retry; exit 2"))
#CHECKGROUP := $(shell dscl . -read /Groups/$(GROUP) >/dev/null 2>/dev/null || (echo "echo Group $(GROUP) nonexistent. Create the group and retry; exit 2"))
```
and they work fine when uncommented.

BTW: these lines are way too old and should be removed from the makefile:
```
# For OSX, comment out the previous two lines and comment in these
#
#CHECKUSER  := $(shell nicl . -read /users/$(USER) >/dev/null 2>/dev/null || (echo "echo User $(USER) nonexistent. Create the user and retry; exit 2"))
#CHECKGROUP := $(shell nicl . -read /groups/$(GROUP) >/dev/null 2>/dev/null || (echo "echo Group $(GROUP) nonexistent. Create the group and retry; exit 2"))
```

**Comment by steveschnepp** [comment]
Well, it's an old issue.

Nowadays I'd just use the current user anyway. As macOS isn't really a server, and therefore it makes much more sense to simply use the currently logged in user.

**Comment by sumpfralle** [comment]
We probably just need an `ifeq($(OSTYPE),foo)` conditional to pick the right set of lines in both Makfiles (`Makefile.conf` and `Makefile.config-maint`).

Can/should we distinguish between older versions of macOS (before 10.5) and newer ones?
Probably we can just assume the newer version for this rare setup.
Would `darwin` be the `OSTYPE` for that platform?

**Comment by steveschnepp** [comment]
I'd think that old macOS versions are way less common than old Linux versions.

**Comment by sumpfralle** [comment]
Fixed in 7b5035985c53ef385f2517fa2ebd616f26c51496.
